src/zope/contentprovider/tests.py

```diff
@@ -20,6 +20,17 @@
 
 from zope.component import eventtesting
 from zope.testing import cleanup, renormalizing
+from zope.contentprovider.interfaces import UpdateNotCalled
+
+
+class TestExceptionHandling(unittest.TestCase):
+
+    def test(self):
+        try:
+            raise UpdateNotCalled
```

Comment:
Would it be more clear that this is testing something if this used `self.assertRaises`? To test the default argument, you could use:
```python
with self.assertRaisesRegexp(UpdateNotCalled, "update() was not called yet."):
    raise UpdateNotCalled
```

Reply:
I believe the default argument is already tested in the README.

I mostly added this to see if Python would give me a syntax error or something for trying to use something that's not a class in the `except:` line.

It would probably be more readable with an assertRaises TBH.  Although, given that the code that raises is right here in the test code, it would still look strange.

Perhaps

```
# Regression test for https://github.com/zopefoundation/zope.contentprovider/issues/4
```

would provide sufficient context to explain why this test is here?  I don't know.  I think my bucket of care is running close to empty....

Reply:
I forget about the readme files...Anyway, not a big deal either way.